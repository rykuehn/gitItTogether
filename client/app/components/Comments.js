import React, { Component } from 'react';
import axios from 'axios';

require('../css/App.css');

class Comments extends Component {
  constructor(props) {
    super(props);

    this.state = {
      comments: [],
    };
  }

  componentDidMount() {
    this.getComments(this.props.url);
  }

  // fetch comments specific to the issue clicked
  getComments(url) {
    axios.get(url)
          .then((response) => {
            this.setState({ comments: response.data });
          });
  }

  render() {
    if (this.state.comments.length > 0) {
      return (
        <div>
          {this.state.comments.map((comment, i) => {
            return (
              <p className="text" key={i}>
                <span><a href={`https://github.com/${comment.user.login}`}>@{comment.user.login}: </a></span>
                {comment.body}
              </p>
            )
          })}
        </div>
      );
    } else {
      return <p className="text"> No Comments </p>;
    }
  }

}

export default Comments;
