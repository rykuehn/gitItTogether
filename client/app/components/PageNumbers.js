import React from 'react';

const PageNumbers = (props) => {
  return (
      <p style={styles.text}>
        {props.num}
      </p>
  )
}

const styles = {
  text: {
    fontFamily: 'PT Sans Narrow',
    border: '1px solid green'
  }
}


export default PageNumbers;