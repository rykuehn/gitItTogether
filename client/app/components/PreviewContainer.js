import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';

import PreviewUser from './PreviewUser';
import PreviewHeading from './PreviewHeading';
import PreviewText from './PreviewText';
import { displayIssue } from '../actions/actions_pages';

const PreviewContainer = (props) => {
  localStorage.setItem(props.issue.id, JSON.stringify({ issue: props.issue }));
  return (
    <Link
      onClick={() => props.displayIssue(props.current, props.position)} style={styles.container}
      to={{
        pathname: 'issueProfile',
        query: { user: props.issue.user.login, IssueID: props.issue.id } }}
    >
      <PreviewHeading issue={props.issue} />
      <PreviewText text={props.issue.body} />
      <PreviewUser user={props.issue.user} />
    </Link>
  );
};

const mapStateToProps = (state) => {
  return {
    current: state.currentPageReducer,
    display: state.displayingReducer,
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ displayIssue }, dispatch);
};

const styles = {
  container: {
    border: '1px solid #D7D9DC',
    borderTop: '0px',
    justifyContent: 'center',
    flexGrow: 1,
    padding: '3vmin',
    textDecoration: 'none',
    color: 'black',
  },
};

export default connect(mapStateToProps, mapDispatchToProps)(PreviewContainer);

