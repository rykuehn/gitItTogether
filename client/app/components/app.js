import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';
import axios from 'axios';

import IssuesListContainer from './IssuesList';
import { issues } from '../actions/actions_pages';

require('../css/App.css');

export class App extends Component {

  componentWillMount() {
    axios.get('https://api.github.com/repos/npm/npm/issues')
         .then((response) => {
           this.props.issues(response.data);
           browserHistory.push('issuesList');
         });
  }
  render() {
    return (
      <div className="body">
        {this.props.totalPages >= 1 ? <IssuesListContainer /> : ''}
        <div>
          {this.props.children}
        </div>
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    totalPages: state.totalPagesReducer,
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ issues }, dispatch);
};


export default connect(mapStateToProps, mapDispatchToProps)(App);
