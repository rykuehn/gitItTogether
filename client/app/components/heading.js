import React from 'react';

require('../css/Heading.css');

const Heading = (props) => {
  return (
    <div>
      <div className="headingContainer">
        <h1 className="headingTitle"> Github Issues </h1>
      </div>
      <div className="color"><div className="arrow" /></div>
      {props.children}
    </div>
  );
};

export default Heading;
