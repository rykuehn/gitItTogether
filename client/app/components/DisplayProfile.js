import React, { Component } from 'react';
import { connect } from 'react-redux';
import Comments from './Comments';


export class ProfileDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      display: null,
    };
  }

  componentWillMount() {
    const ID = window.location.href.split('?')[1].split('&')[0].split('=')[1];
    this.setState({ display: JSON.parse(localStorage.getItem(ID)) });
  }

  render() {
    const { total, current, display, pages } = this.props;
    const issue = total >= 1 ? pages[current][display] : this.state.display.issue;

    return (
      <div style={styles.container}>
        <div>
          <div> {issue.title} </div>
          <div> {issue.id} </div>
          <div> {issue.state} </div>
        </div>
        <div style={styles.body}>{issue.body} </div>
        <div>@<a href={`https://github.com/${issue.user.login}`}>{issue.user.login}</a></div>
        <div className="labels">
          {issue.labels.map((label, i) => {
            return <div key={i}> {label.name} </div>;
          })}
        </div>
        <img alt="user avatar" style={styles.image} src={issue.user.avatar_url} />
        <Comments url={issue.comments_url} />
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    pages: state.pagesReducer,
    current: state.currentPageReducer,
    display: state.currentIssueDisplayReducer,
    total: state.totalPagesReducer,
  };
};

const styles = {
  container: {
    display: 'flex',
    width: '70vw',
    border: '1px solid blue',
    flexDirection: 'column',
    alignItems: 'center',
  },
  body: {
    border: '1px solid green',
  },
  image: {
    width: '10vmin',
    height: '10vmin',
  },
};


export default connect(mapStateToProps, null)(ProfileDetails);
