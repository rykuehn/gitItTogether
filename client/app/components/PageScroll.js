import React from 'react';
import _ from 'underscore';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { currentPage } from '../actions/actions_pages';
import PageNumbers from './PageNumbers';

const PageScroll = (props) => {
  var amount = _.range(1, props.pageNums + 1);
  return (
      <div style={styles.container}>
        {amount.map(number => {
          return <PageNumbers click={props.currentPage} key={number} num={number} />
        })}
      </div>
  )
}


const mapDispatchToProps = dispatch => { 
  return bindActionCreators({ currentPage }, dispatch);
}

const styles = {
  container: {
    border: '1px solid black',
    height: '10vmin',
    width: '30vmin',
    display: 'flex'
  }
}


export default connect(null, mapDispatchToProps)(PageScroll);