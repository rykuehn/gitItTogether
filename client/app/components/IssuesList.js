import React, {Component} from 'react';
import PreviewContainer from './PreviewContainer';

class IssuesList extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
      <div style={styles.container}>
        {this.props.issues.slice(0, 25).map(issue => {
          return <PreviewContainer issue={issue} key={issue.id} />
        })}
      </div>
    )
    
  }
}

const styles ={
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    width: '130vmin',
    border: '1px solid red',
    justifyContent: 'flex-start',
    alignItems: 'space-around',
    flexDirection: 'column'
  }
}
export default IssuesList;