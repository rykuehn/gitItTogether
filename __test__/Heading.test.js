import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Heading } from '../client/app/components/heading';

it('Heading renders correctly', () => {
  const tree = renderer.create(<Heading />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('shallow render of Heading Componenet', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Heading />);
  });

  it('contains header title Issues', () => {
    expect(component.contains(<h1 className="headingTitle"> Github Issues </h1>)).toBe(true);
  });
});
