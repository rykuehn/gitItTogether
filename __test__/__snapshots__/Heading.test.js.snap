// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<div>
  <div
    className="headingContainer"
  >
    <h1
      className="headingTitle"
    >
       Github Issues 
    </h1>
  </div>
  <div
    className="color"
  >
    <div
      className="arrow"
    />
  </div>
</div>
`;
